### ISSUE TITLE
VERSION =~ /_/ forces -TRIAL in tarball name.

### ISSUE THREAD

**Issue by kentfredric** [rjbs/Dist-Zilla#475]
An annoying consequence of chained behaviour:

- [When version =~ /_/, release_status defaults to "testing"](https://metacpan.org/source/RJBS/Dist-Zilla-5.037/lib/Dist/Zilla/Dist/Builder.pm#L425)
- [When release_status is "testing", is_trial defaults to true](https://metacpan.org/source/RJBS/Dist-Zilla-5.037/lib/Dist/Zilla.pm#L550)
- [When is_trial is true, '-TRIAL' is always appeneded.](https://metacpan.org/source/RJBS/Dist-Zilla-5.037/lib/Dist/Zilla/Dist/Builder.pm#L425)

All this means the only way to force `is_trial` to be **false** when using a `_` version, you have to set the release status to a stable one ....

This is pretty much a "Go away" from Dzil.

**Comment by genehack** [comment]
Even trying to explicitly set a stable release doesn't work:

```
% RELEASE_STATUS=stable V=0.045_01 dzil build                                                                 fuck-versioning ✗ ✹ 10:11:02
[Run::BeforeBuild] executing: rm -f Makefile.PL
[DZ] beginning to build Git-Wrapper
[@Filter/ExtraTests] rewriting author test xt/author/err.t
[@Filter/ExtraTests] rewriting release test xt/release/pod-syntax.t
[@Filter/ExtraTests] rewriting release test xt/release/pod-coverage.t
[DZ] guessing dist's main_module is lib/Git/Wrapper.pm
[DZ] writing Git-Wrapper in Git-Wrapper-0.045_01
[@Filter/MetaYAML] Invalid META structure.  Errors found:
[@Filter/MetaYAML] 'stable' for 'release_status' is invalid for version '0.045_01' (release_status) [Validation: 2]
[@Filter/MetaYAML] Invalid META structure.  Errors found:
[@Filter/MetaYAML] 'stable' for 'release_status' is invalid for version '0.045_01' (release_status) [Validation: 2] at /opt/plenv/versions/5.22.0/lib/perl5/site_perl/5.22.0/darwin-2level/Moose/Meta/Method/Delegation.pm line 110.
```

**Comment by genehack** [comment]
FWIW, this did not happen with 5.031. @ether++ for the pointer.

**Comment by karenetheridge** [comment]
Setting release_status=stable would definitely be wrong, as then the metadata will be inconsistent with what the version is telling us -- I don't think PAUSE would ever consent to indexing a version with an underscore in it.

I'd support simply removing -TRIAL from the archive name when the version contains an underscore, as in this case it's redundant information. I'll PR. :)

**Comment by genehack** [comment]
**Update update:** 5.036 goes back to working, but adds the '-TRIAL'.

**Comment by karenetheridge** [comment]
```
5.036     2015-05-02 11:08:51-04:00 America/New_York
        - BUGFIX: detection of trial status via underscore in version was
          accidentally lost in v5.035; restored now!

5.035     2015-04-16 15:43:26+02:00 Europe/Berlin
        - BREAKING CHANGE: is_trial is now read-only
        - release_status is a new Dist::Zilla attribute and
          ReleaseStatusProvider plugin role
```

**Comment by rjbs** [comment]
This appears solved by 646efa80e8f13333b82495e6990b2c6db43f28b5.
